<launch>
	<group ns="platform_executors">
		<node pkg="line_action_server" name="lineActionServer" type="lineGoalActionServer.py" output="screen">
			<param name="cmd_vel_topic" value="/fmSignals/cmd_vel" />
			
			<param name="odometry_topic" value="/odom" /> <!-- Topic for receiving odometry messages if tf is not used -->
			<param name="odom_frame" value="/odom" /> <!-- Odometry frame if tf is used -->
			<param name="base_frame" value="/odom" /> <!-- Base frame if tf is used -->
			<param name="use_tf" value="False"/> <!-- Choose tf or odometry topic -->
			
			<param name="max_linear_velocity" value="0.4"/> <!-- maximum linear velocity in autonomous mode in [m/s]-->
			<param name="max_angular_velocity" value="1.8"/> <!-- maximum angular velocity in autonomous mode in [rad/s]-->
			<param name="max_distance_error" value="0.05"/> <!-- radius of the target in [m]-->
			<param name="max_distance_from_line" value="0.3/> <!-- At this distance from line the robot will hold and correct in [m] -->
			<param name="retarder" value="0.3"/> <!-- Factor for reducing speed in difficult situations -->
			<param name="max_angle_error" value="0.78"/> <!-- Maximum allowed angle error (Adaptive for line controller) [rad]-->
			<param name="max_initial_error" value="0.17"/> <!-- Maximum allowed angle error at initialisation in [rad] -->
			
			<param name="period" value="0.1"/> <!-- sets the period of the control loop in [s] -->
			<param name="lin_p" value="0.4"/> <!-- Proportional gain for linear velocity controller -->
			<param name="lin_i" value="0.6"/> <!-- Integral gain for linear velocity controller -->
			<param name="lin_d" value="0.0"/> <!-- Differential gain for linear velocity controller -->
			<param name="ang_p" value="0.8"/> <!-- Proportional gain for angular velocity controller -->
			<param name="ang_i" value="0.1"/> <!-- Integral gain for angular velocity controller -->
			<param name="ang_d" value="0.05"/> <!-- Differential gain for angular velocity controller -->
			<param name="integrator_max" value="0.1"/> <!-- Common anti-windup max for integrators -->
		</node>
	</group>	
</launch>